src/persist-client/src/internal/trace.rs

```diff
@@ -103,10 +112,225 @@ impl<T: Timestamp + Lattice> Default for Trace<T> {
     fn default() -> Self {
         Self {
             spine: Spine::new(),
+            roundtrip_structure: false,
         }
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ThinSpineBatch<T> {
+    pub(crate) level: usize,
+    pub(crate) desc: Description<T>,
+    pub(crate) parts: Vec<SpineId>,
+}
+
+#[derive(Clone, Debug)]
+pub struct FlatTrace<T> {
+    pub(crate) since: Antichain<T>,
+    pub(crate) legacy_batches: BTreeMap<Arc<HollowBatch<T>>, ()>,
+    pub(crate) hollow_batches: BTreeMap<SpineId, Arc<HollowBatch<T>>>,
+    pub(crate) spine_batches: BTreeMap<SpineId, ThinSpineBatch<T>>,
```

Comment:
Also as mentioned offline, we have a nice space here to hang metadata about a fueling compaction. Will be nice to have the same for a fueled, outstanding compaction request (e.g. when it started, checkpoint incremental progress, etc). How do you see that fitting into this?

Reply:
Discussed elsewhere: we've decided that attaching this kind of metadata to the merge itself seems natural, and since a merge is already associated with a spine id, we shouldn't need a new collection for it hopefully.